src/mewbot/tools/__init__.py

```diff
@@ -123,49 +134,128 @@ def run_tool(self, name: str, *args: str) -> subprocess.CompletedProcess[bytes]:
         arg_list = list(args)
         arg_list.extend(self.folders)
 
-        run_result = self._run_utility(name, arg_list)
+        run_result = self.loop.run_until_complete(self._async_run_utility(name, arg_list))
+
+        self.success = self.success and (run_result.returncode == 0)
+        assert isinstance(run_result, subprocess.CompletedProcess)
+        return run_result
+
+    async def async_run_tool(
+        self, name: str, *args: str
+    ) -> subprocess.CompletedProcess[bytes]:
+        """
```

Comment:
Don't duplicate the logic here, just have `run_tool` be
```
return self.loop.run_until_complete(self.async_run_tool(name, *args)
```

Or just make `run()` and `run_tool()` always async, move the loop logic up into `__call__`, and that means that tools can easily set up their own concurrency.